tests/drivers/adc/adc_api/src/test_adc.c

```diff
@@ -18,8 +20,18 @@
 #define INVALID_ADC_VALUE SHRT_MIN
 #endif
 
+#if CONFIG_NOCACHE_MEMORY
+#define __NOCACHE	__attribute__((__section__(".nocache")))
+#else /* CONFIG_NOCACHE_MEMORY */
+#define __NOCACHE
+#endif /* CONFIG_NOCACHE_MEMORY */
+
 #define BUFFER_SIZE  6
+#ifdef CONFIG_TEST_USERSPACE
 static ZTEST_BMEM int16_t m_sample_buffer[BUFFER_SIZE];
```

Comment:
No need to define this memory as non cacheable ?

Reply:
No. The case with CONFIG_TEST_USERSPACE is for the basic adc_api test, without DMA. It has worked without like this for a long time, and setting nocache attribute with USERSPACE enabled causes MPU fault anyway.
For the test with DMA, we have to disable USERSPACE to define the nocache attribute.